### PR TITLE
manifest: Minimize host;etcd/flannel/kube/storage as layers/containers

### PR DESCRIPTION
**Pull request by cgwalters**
:construction: Do not merge :construction:  - pushing for comments/early review.

### CODE REVIEW

**Comment by cgwalters** [comment]
Pushing to share, not yet for merge :construction:

**Comment by cgwalters** [comment]
TL;DR architecture proposal:

 - kubelet/kube-proxy via RPM
 - etcd/flannel via system containers
 - master/apiserver/scheduler etc. via Docker

Collection of links/work on cluster bootstrap:

kubelet (as layered RPM package):

 - `rpm-ostree pkg-add kubernetes-node`

etcd/flannel (as system containers):

 - https://github.com/yuqi-zhang/atomic-system-containers-quickstart

kubernetes

 - https://jebpages.com/2016/08/15/running-kubernetes-in-containers-on-atomic/
 - https://access.redhat.com/documentation/en/red-hat-enterprise-linux-atomic-host/version-7/getting-started-with-containers/#creating_a_kubernetes_cluster_to_run_docker_formatted_container_images
 - http://kubernetes.io/docs/getting-started-guides/scratch/#configuring-and-installing-base-software-on-nodes

**Comment by miabbott** [comment]
Is there related story/docs for containerized ceph/gluster?

**Comment by cgwalters** [comment]
Yeah, storage :file_folder: is another thread, thanks for bringing that up. I did verify that `pkg-add` works to re-add the packages we have now.  Will edit the above comment.

**Comment by jasonbrooks** [comment]
Will we build/deliver the containers from: https://wiki.centos.org/ContainerPipeline

**Comment by cgwalters** [comment]
Yeah we should.  There are some in https://github.com/fedora-cloud/Fedora-Dockerfiles/pull/112 which we either need to progress or fork into centos or so.

**Comment by cgwalters** [comment]
There's still a number of open design items - we could e.g. bake in kubelet rather than layer via RPM.  Some users may choose layered packages over syscontainers.  We'll see.  Let's move forward with this for now in Alpha and build on it.
